AzureAppServiceSetAppSettings/Tasks/AppSettings/ApplyAppSettings.ps1

```diff
@@ -29,7 +36,11 @@ $splitOption = [System.StringSplitOptions]::RemoveEmptyEntries
 $lines = $AppSettings.Split($seperator, $splitOption)
 Write-Host ("Lines found: " + $lines.Count)
 
-$webApp = Get-AzureRMWebApp -Name $WebAppName -ResourceGroupName $ResourceGroupName
+$webApp = If ($useSlot) {
+    Get-AzureRMWebAppSlot -Name $WebAppName -ResourceGroupName $ResourceGroupName -Slot $Slot
+} Else { 
+    Get-AzureRMWebApp -Name $WebAppName -ResourceGroupName $ResourceGroupName
```

Comment:
you don't have to do this if, you could use the 'production' slot instead and only rely on Get-AzureRMWebAppSlot